orchestrator/dags/general_data_import.py

```diff
@@ -102,30 +101,29 @@ def extract_timetable() -> List[TimeTable]:
     for file in os.listdir(out_dir):
         filename = os.fsdecode(file)
         if filename.endswith(".txt") and filename.startswith("tram"):
-            f = open(f'{out_dir}{filename}', "r", encoding="utf8")
-            for line in f:
-                if 'Linia:' in line:
-                    line_number = line.split()[1]
-                elif '*TR' in line:
-                    num_of_route = line.strip()[3:]
-                elif '*RP' in line or '#OP' in line:
-                    line = next(f)
-                    stop_info = line.split()
-                    unit = stop_info[0][0:4]
-                    post = stop_info[0][4:]
-                elif '*OD' in line:
-                    i = int(line.split()[1])
-                    line = next(f)
-                    for n in range(i):
-                        hour, stop = line.split()
-                        departure_time = convert_to_time(hour)
-                        route, day_type, _ = stop.split('/')
-                        new_time_table = TimeTable(int(line_number), route, day_type, unit, post, str(departure_time), n)
-                        time_table.append(new_time_table)
+            with open(f'{out_dir}{filename}', "r", encoding="utf8") as f:
+                for line in f:
+                    if 'Linia:' in line:
+                        line_number = line.split()[1]
+                    elif '*TR' in line:
+                        num_of_route = line.strip()[3:]
+                    elif '*RP' in line or '#OP' in line:
                         line = next(f)
-                else:
-                    continue
-            f.close()
+                        stop_info = line.split()
+                        unit = stop_info[0][0:4]
+                        post = stop_info[0][4:]
+                    elif '*OD' in line:
+                        i = int(line.split()[1])
+                        line = next(f)
```

Comment:
Here, and in lines 111 and 124 the `line` variable is reassigned to refer next line from file. I guess these lines should be wrapped in `try/except StopIteration` block.